### PR TITLE
Make Worldwide taxons expand/collapse per country

### PR DESCRIPTION
**Pull request by 1pretz1**
https://trello.com/c/O7if1mGu/147-improve-the-worldwide-tagging-interface-3

Currently, only `world` expands which displays a huge list of all
the countries and their sub taxons. This is not good for the user as its
an overload of information.

This change provides a link to expand/collapse per country, this new list
also displays everything alphabetically starting with all the taxons
that have no children.

## Before 
worldwide-tagging

## After
screen shot 2018-06-13 at 12 33 19

### CODE REVIEW

File: lib/taxonomy/world_taxonomy.rb [hunk 1]
```diff
@@ -6,11 +6,17 @@ def initialize(adapter: RedisCacheAdapter.new, tree_builder_class: Tree)
     end
 
     def all_world_taxons
-      @_world_taxon_list ||= world_taxon_branches
+      @_all_world_taxons ||= sort_world_taxons
     end
 
   private
 
+    def sort_world_taxons
```
Comment (cbaines) [inline lib/taxonomy/world_taxonomy.rb]: I'd expect `sort_world_taxons` to take some taxons, and sort them. Maybe `sorted_world_taxons` would be a better name for the way this is being used?

File: test/unit/taxonomy/world_taxonomy_test.rb [hunk 2]
```diff
@@ -7,14 +7,15 @@ def subject
...
+  test "#sort world_taxons alphabetically when children == 0, then alphabetically on world_taxons with children" do
     redis_cache_has_world_taxons(
       [
-        build(:taxon_hash, content_id: 'content_id_1'),
-        build(:taxon_hash, content_id: 'content_id_2'),
-        build(:taxon_hash, content_id: 'content_id_3')
+        build(:taxon_hash, title: 'France', children: [child_taxon]),
+        build(:taxon_hash, title: 'Australia', children: [child_taxon, child_taxon]),
+        build(:taxon_hash, title: 'Vietnam'),
+        build(:taxon_hash, title: 'Germany')
```
Comment (cbaines) [inline test/unit/taxonomy/world_taxonomy_test.rb]: I'd use more realistic data here, like a title of "News and Events", just to help people in the future make sense of the tests.

File: test/unit/taxonomy/world_taxonomy_test.rb [hunk 3]
```diff
@@ -7,14 +7,15 @@ def subject
     Taxonomy::WorldTaxonomy.new
   end
 
-  test "#world_taxons" do
+  test "#sort world_taxons alphabetically when children == 0, then alphabetically on world_taxons with children" do
```
Comment (cbaines) [inline test/unit/taxonomy/world_taxonomy_test.rb]: This doesn't quite make sense to me, perhaps just say `test "#order of world taxons"`?

File: test/support/taxonomy_helper.rb [hunk 4]
```diff
@@ -134,15 +134,15 @@ def draft_taxon_2
 
   def world_grandchild_taxon
     FactoryBot.build(:taxon_hash,
-                     title: "World Child Taxon",
+                     title: "World grand child taxon",
```
Comment (cbaines) [inline test/support/taxonomy_helper.rb]: grandchild is one word, not two.

**Comment by cbaines** [comment]
It looks like you've made some changes related to comments I've made. Just a note, but it is useful to comment and say this has happened, just to make it clear what is going on.

**Comment by cbaines** [comment]
Is this change going to require the cache for the taxon data to be repopulated from the Publishing API?

**Comment by 1pretz1** [comment]
Yes, the cache would have to be refreshed!

File: test/integration/world_tagging_flow.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    it "asserts WorldWide tagging path" do
+      visit_world_tags_path
+      assert_equal(200, page.status_code)
+    end
+
+    it "asserts a selected checkbox' " do
+      visit_world_tags_path
+      expand_and_check_checkbox
+      assert page.has_checked_field?('World grand child taxon')
```
Comment (cbaines) [inline test/integration/world_tagging_flow.rb]: I'm a bit surprised this is still working, as the taxon has now been changed to "grandchild"...

Comment (1pretz1) [reply test/integration/world_tagging_flow.rb]: Urmm, weird! I've ran it locally and it failed, i've corrected the test accordingly.

**Pull request merged**